packages/connect-react/src/index.ts

```diff
@@ -1,7 +1,162 @@
+  createElement,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import {
+  App,
+  ConnectOptions,
+  ConnectorDeclaration,
+  Organization,
+  Permission,
+  connect,
+} from '@aragon/connect'
+
+type ConnectProps = {
+  children: React.ReactNode
+  connector: ConnectorDeclaration
+  location: string
+  options?: ConnectOptions
+}
+type ContextValue = {
+  org: Organization | null
+  orgLoadingStatus: LoadingStatus
+}
+type LoadingStatus = {
+  error: Error | null
+  loading: boolean
+  retry: () => void
+}
+type OrganizationHookResult = [
+  Organization | null,
+  { error: Error | null; loading: boolean; retry: () => void }
+]
+type AppsHookResult = [App[], LoadingStatus]
+type PermissionsHookResult = [Permission[], LoadingStatus]
+
+const ConnectContext = React.createContext<ContextValue>({
+  org: null,
+  orgLoadingStatus: { error: null, loading: false, retry: () => {} },
+})
+
+export function Connect({
+  children,
+  connector,
+  location,
+  options,
+}: ConnectProps): JSX.Element {
+  const [org, setOrg] = useState<Organization | null>(null)
+  const [orgError, setOrgError] = useState<Error | null>(null)
+  const [orgLoading, setOrgLoading] = useState<boolean>(false)
+
+  const cancelOrgLoading = useRef<Function | null>(null)
+
+  const loadOrg = useCallback(() => {
+    let cancelled = false
+
+    cancelOrgLoading.current?.()
+    cancelOrgLoading.current = () => {
+      cancelled = true
+    }
+
+    setOrgLoading(true)
+
+    const update = async () => {
+      const done = (err: Error | null, org: Organization | null) => {
+        if (!cancelled) {
+          setOrgLoading(false)
+          setOrgError(err || null)
+          setOrg(err ? null : org)
+        }
+      }
+
+      try {
+        const org = await connect(location, connector, options)
+        done(null, org)
+      } catch (err) {
+        done(err, null)
+      }
+    }
+    update()
+  }, [location, connector, options])
+
+  useEffect(loadOrg, [location, connector, options])
+
+  const value = useMemo<ContextValue>(
+    () => ({
+      org,
+      orgLoadingStatus: {
+        error: orgError,
+        loading: orgLoading,
+        retry: loadOrg,
+      },
+    }),
+    [org, orgError, orgLoading, loadOrg]
+  )
+
+  return createElement(ConnectContext.Provider, { value, children })
 }
 
-export function AragonConnector(): void {
-  return
+export function useOrganization(): OrganizationHookResult {
+  const { org, orgLoadingStatus } = useContext<ContextValue>(ConnectContext)
+  return [org, orgLoadingStatus]
 }
+
+function useConnectSubscription<Data>(
+  cb: (
+    org: Organization,
+    onData: (data: Data) => void
+  ) => { unsubscribe: Function },
+  initValue: Data
+): [Data, LoadingStatus] {
+  const [data, setData] = useState<Data>(initValue)
+  const [loading, setLoading] = useState<boolean>(false)
+  const { org } = useContext<ContextValue>(ConnectContext)
+  const cbRef = useRef(cb)
```

Comment:
I think this was discussed earlier (https://github.com/aragon/connect/pull/52#discussion_r448598635), but the API feels kind of awkward to enforce that "your callback may never change".

Otherwise, it might be worth adding a comment to explain that this hook expects its parameters to never change.

Reply:
Yes you’re right, we might want to change that. Especially now that we have parameters on `useApps()` and `useApp()`.

Reply:
Done!